### PR TITLE
feat: provide annotation file for genes

### PR DESCRIPTION
**Pull request by Hao-Chalmers**
### Main improvements in this PR:
This PR is to address the issue discussed in #157
* a `JSON` file with gene annotation info is provided by extracting relevant associations from `ensemble_ID_mapping.tsv`
* the `basicCheck` script was adapted by adding the consistency check for genes, and renamed to `sanityCheck`

**I hereby confirm that I have:**

- [X] Tested my code on my own computer for running the model
- [X] Selected `devel` as a target branch

### CODE REVIEW

**Review by mihai-sysbio — COMMENTED** [review]
I understand the intention was to not approach #203 in this PR. My personal approach would have been to skip #157 and solve #203 directly.
Regardless, I'm wondering why does the annotation files need to be checked routinely, rather than them being re-created with every new release.

**Comment by Hao-Chalmers** [comment]
> I'm wondering why does the annotation files need to be checked routinely, rather than them being re-created with every new release.

Recreating `tsv` format annotation files with every release might be a good solution to #203

**Comment by JonathanRob** [comment]
@mihai-sysbio

> Regardless, I'm wondering why does the annotation files need to be checked routinely, rather than them being re-created with every new release.

I'm not sure I understand how one would "re-create" these files, since the information in those files is not stored anywhere else (except the gene annotations).

@Hao-Chalmers 

> Recreating tsv format annotation files with every release might be a good solution to #203.

Just to clarify, are you suggesting that tsv format files are generated _in addition to_ the JSON files, or that they will _replace_ the JSON files?

Also a very minor point, but why not just name the script based on what it is doing? i.e., `consistencyCheck`.

**Comment by Hao-Chalmers** [comment]
Since this PR aims to address #157, I'll post responses to `tsv` annotation files under #203 for better tracking of the information flow.

**Comment by mihai-sysbio** [comment]
@JonathanRob regarding the re-creation of these annotation files: somebody somehow generates them (which by the way is not really transparent how), and then they are routinely checked for consistency. Instead of that, I'm suggesting to automate the generation of these files, and to thus skip the consistency check. This doesn't need to be implemented here; #203 would be a good place in my opinion.

**Review by mihai-sysbio — APPROVED** [review]
A tiny comment: the branch name doesn't follow any particular convention.

**Comment by JonathanRob** [comment]
@mihai-sysbio I think that's the problem - there is no way to "re-generate" these files, as far as I know. They were originally created by taking information that was stored in the model, and moving it to a different format (`JSON`) (i.e., deleted from the model). These annotation JSON files were then curated over time in parallel with the model. So, as far as I know, these JSON files are the **only** source of this information, and therefore we cannot re-generate them from some other source.

However, this could partially change in the future, if we decide to source some of the annotations from other databases, such as MetaNetX.

**Comment by Hao-Chalmers** [comment]
@mihai-sysbio regarding the branch, which is often to address particular issues listed, if it's a good idea to directly name it by the issue number?

File: code/test/sanityCheck.py [hunk 1]
```diff
@@ -12,20 +12,10 @@ def load_yml(yml_file):
...
-        modelMets.append(m.id)
-
-    # get list of genes
-    modelGenes = []
-    for g in model.genes:
-        modelGenes.append(g.id)
+    # get lists of rxns, mets and genes
+    modelRxns  = list(map(lambda element : element.id, model.reactions))
+    modelMets  = list(map(lambda element : element.id, model.metabolites))
+    modelGenes = list(map(lambda element : element.id, model.genes))
```
Comment (mihai-sysbio) [inline code/test/sanityCheck.py:18]: Nice! How about reusing the _lambda_ function like so:
> fetchId = lambda x : x.id
extractIds = lambda componentArray : list(map(fetchId, componentArray))
[ modelRxns, modelMets, modelGenes] = list(map(extractIds , [model.reactions, model.metabolites, model.genes] ))

Disclaimer: the above might need some testing.

**Comment by mihai-sysbio** [comment]
@Hao-Chalmers In MetAtlas we tend to follow the same style as for the semantic commits. In this particular case, I would have probably gone for something like `feat/gene-annotation_157`.

File: code/test/sanityCheck.py (hunk 1 above)
Comment (Hao-Chalmers) [reply code/test/sanityCheck.py:18]: this idea was actually considered, but seems against simplicity and readability rules.

Comment (mihai-sysbio) [reply code/test/sanityCheck.py:18]: I see. The first bit though (`fetchId = lambda x : x.id`) should increase readability, unless you expect some variation in the future in how the id will be stored across different component types.

**Pull request merged**